---
title: "Group04 Presentation R4BDS"
author: "Eric Calaveras - Sol Escamilla - Beatriz García - Lucía López - Diego Medina"
format: revealjs
editor: visual
css: styles.css
echo: True
---

## Introduction

<div>

<br>

</div>

## Materials and methods

#### Data Wrangling: From raw to tidydata

<br>

------------------------------------------------------------------------

layout: 2-col

------------------------------------------------------------------------

::: column
```{r eval=FALSE}

borovecki <- borovecki$x |>
  as_tibble() |>
  mutate(
    sample = row_number(), 
    .before = 1, 
    diagnosis = borovecki$y
  )

borovecki <- borovecki |>
  pivot_longer(
    cols = ends_with("at"),
    names_to = "probeID", 
    values_to = "expressionValue",
    values_drop_na = TRUE
  )

```
:::

::: column

Imagen de la tibble
:::

## Materials and methods

#### Data Wrangling: From to tidydata

---
layout: 2-col
---

::: column
```{r}
#| message = FALSE
borovecki <- borovecki |>
  mutate(
    gene = mapIds(
      hgu133a.db, 
      keys=probeID, 
      keytype = "PROBEID", 
      column="SYMBOL",
      multiVals="first"
    )
  ) |>
  filter(!is.na(gene)) |>
  # I specify the package because AnnotationDbi also has a select function
  dplyr::select(-probeID)

borovecki_geneLevel <- borovecki |>
  group_by(sample, gene) |>
  summarise(expression = mean(expression), .groups = "drop")

```
:::

::: column
imagen de la tibble
:::

## Materials and methods

#### Normalization: From messy to tiddy

![](images/comparison1_raw_log2-05.png){fig-align="center"}

## Materials and methods

#### Linear Modelling 

---
layout: 2-col
---

::: column
```{r}
borovecki_tibble <- borovecki_tibble |>
  pivot_wider(
    # Columns not to pivot
    id_cols = c(sample, diagnosis),
    # Columns to pivot
    names_from = gene,
    values_from = expressionValue
  ) |>
  mutate(
    Diagnosis = case_when(
      diagnosis == "symptomatic" ~ 1,
      diagnosis == "control" ~ 0
    )
  ) |>
  dplyr::select(-c(diagnosis,sample)) |>
  relocate(Diagnosis, .before = 1)

borovecki_tibble_long <- borovecki_tibble  |> 
  pivot_longer(
    cols = -Diagnosis,
    names_to = "gene",
    values_to = "log2_expr_level"
  ) |>
  mutate(
    log2_expr_level = log2(log2_expr_level)
  )
borovecki_tibble_long_nested <- borovecki_tibble_long |> 
  group_by(gene) |> 
  nest() |> 
  ungroup()
```
:::

::: column
imagen de la tibble
:::

## Materials and methods

#### Linear Modeling

<br>

---
layout: 2-col
---

::: column
```{r}
by_gene <- borovecki_tibble_long_nested |> 
  group_by(gene)
# Define lm function
linear_model <- function(df) {
  lm(
  log2_expr_level ~ Diagnosis , 
  data = df)
}
# Apply lm function to each gene
borovecki_tibble_long_nested  <- borovecki_tibble_long_nested  |>  
  mutate(model_object = map(data, linear_model))
```
:::

::: column
imagen
:::

## Results

<br>

![](images/Expression_of_DNA_repair_genes_vs_clinical_outcome.png){fig-align="center"}

## Results

<br>

![](images/forest-02.png){fig-align="center"}

## Discussion

<br>

The use of different functions in RStudio has allowed us to modify the initial raw dataset making it tidy so that we could work with it. By doing that, it became easier for us to see the genes that were more differentially expressed between patients with HD and control. Also, bibliography reading was used to get information about the genes implied in the development of the disease. Combining our tidy data with the bibliography, allowed us to see the difference in the expressions by applying different plots.

<br>

This methodology could also be translated for other diseases in the future, creating bonds between hospitals and bioinformaticians to use the data they obtain to give us a better understanding of how different diseases work and express themselves by regulating their gene expression.
