---
title: "Group04 Presentation R4BDS"
author: "Eric Calaveras - Sol Escamilla - Beatriz García - Lucía López - Diego Medina"
editor: visual
echo: true

format:
  revealjs:
    css: styles.css
    background-color: "#CCE6FF"
---

## Introduction

<div>

<br>

![](images/Borovecki$x-01.png){fig-align="center"}

</div>

## Materials and methods

#### Data Wrangling: From messy to tidy data

<br>

---
layout: 2-col
---

::: column
```{r eval=FALSE}

borovecki <- borovecki$x |>
  as_tibble() |>
  mutate(
    sample = row_number(), 
    .before = 1, 
    diagnosis = borovecki$y
  )

borovecki <- borovecki |>
  pivot_longer(
    cols = ends_with("at"),
    names_to = "probeID", 
    values_to = "expressionValue",
    values_drop_na = TRUE
  )
```
:::

## Materials and methods

#### Data Wrangling: From messy to tidy data

---
layout: 2-col
---

::: column
```{r eval=FALSE}

borovecki <- borovecki |>
  mutate(
    gene = mapIds(
      hgu133a.db, 
      keys=probeID, 
      keytype = "PROBEID", 
      column="SYMBOL",
      multiVals="first"
    )
  ) |>
  filter(!is.na(gene)) |>
  dplyr::select(-probeID)

borovecki_geneLevel <- borovecki |>
  group_by(sample, gene) |>
  summarise(expression = mean(expression), .groups = "drop")
```
:::

## Materials and methods

#### Normalization: From raw to standardize data

---
layout: 2-col
---

::: column
```{r eval=FALSE}
raw_histogram <- ggplot(borovecki_data, aes(x = expressionValue,fill = "Raw data")) +
  geom_histogram(bins = 50, alpha = 0.6) +
  fill_scale+
  theme_minimal(base_size = 14) +
  labs(x = "Expression value",
       y = "Count") +
  theme(plot.title = element_text(size = 13, 
                                  hjust = 0.5)) 

log2_histogram <- ggplot(borovecki_data, aes(x = log2_expression,  fill = "Log2 data")) +
  geom_histogram(bins = 100, alpha = 0.6) +
  fill_scale +
  theme_minimal(base_size = 14) +
  labs(x = "log2(Expression value)",
       y = "Count") +
  theme(plot.title = element_text(size = 13,
                                  hjust = 0.5))
raw_histogram + log2_histogram + 
  plot_layout(guides = "collect")+
  plot_annotation(title="Comparison between Raw and Log2-Transformed Expression Data", 
                  caption="Data from: Borovecki et al. 2005") &
  theme(legend.position = "right")
```
:::

![](images/comparison1_raw_log2-05.png){fig-align="center"}

## Materials and methods

#### Linear Modelling

<br>

---
layout: 2-col
---

::: column
```{r eval=FALSE}
borovecki_tibble <- borovecki_tibble |>
  pivot_wider(
    # Columns not to pivot
    id_cols = c(sample, diagnosis),
    # Columns to pivot
    names_from = gene,
    values_from = expressionValue
  ) |>
  mutate(
    Diagnosis = case_when(
      diagnosis == "symptomatic" ~ 1,
      diagnosis == "control" ~ 0
    )
  ) |>
  dplyr::select(-c(diagnosis,sample)) |>
  relocate(Diagnosis, .before = 1)

borovecki_tibble_long <- borovecki_tibble  |> 
  pivot_longer(
    cols = -Diagnosis,
    names_to = "gene",
    values_to = "log2_expr_level"
  ) |>
  mutate(
    log2_expr_level = log2(log2_expr_level)
  )
borovecki_tibble_long_nested <- borovecki_tibble_long |> 
  group_by(gene) |> 
  nest() |> 
  ungroup()
```
:::

::: column
![](images/Borovecki_tibble.png){fig-align="center" width="500"}

![](images/borovecki__tibble_long.png){fig-align="center" width="400"}

![](images/borovecki_tibble_long_nested_prelm.png){fig-align="center" width="300"}
:::

## Materials and methods

#### Linear Modeling

<br>

---
layout: 2-col
---

::: column
```{r eval=FALSE}
by_gene <- borovecki_tibble_long_nested |> 
  group_by(gene)
linear_model <- function(df) {
  lm(
  log2_expr_level ~ Diagnosis , 
  data = df)
}
borovecki_tibble_long_nested  <- borovecki_tibble_long_nested  |>  
  mutate(model_object = map(data, linear_model))
tidy_data <- function(df) {
  df |>
  tidy(conf.int = TRUE,
       conf.level = 0.95)
  }
borovecki_estimates <- borovecki_tibble_long_nested |>
  mutate(model_object_tidy = map(model_object, tidy_data)) |>
  unnest(model_object_tidy) |>
  filter(term == "Diagnosis") |>
  dplyr::select(gene, p.value, estimate, conf.low, conf.high) |> 
  mutate(q.value=p.adjust(p.value),
          is_significant = case_when(
            q.value >= 0.05 ~ "no",
            q.value < 0.05 ~ "yes"
            ),
         lbl = case_when(
           q.value < 0.05 ~ gene,
           q.value >= 0.05 ~ "")
         )
```
:::

::: column
![](images/borovecki_tibble_long_nested.png){fig-align="right"}

![](images/borovecki_estimate.png){fig-align="right"}
:::

## Results

#### Dataset exploration in established genetic modifiers

<br>

---
layout: 2-col
---

::: column
```{r eval=FALSE}
geneExpression_comparison_boxplot <- ggplot(borovecki_grouped |>
  filter(gene %in% c("HTT","FAN1", "LIG1", "MLH1", "MSH3", "PMS1", "PMS2")), aes(gene, log2_expression, fill = diagnosis)) +
  geom_boxplot() +
  theme_minimal() +
  labs(
    title = "Gene expression values in control and symptomatic individuals",
    x = "Genes",
    y = "Genes expression"
  ) +
  theme(
    plot.title = element_text(size = 12)) 

```
:::

::: column
![](images/Expression_of_DNA_repair_genes_vs_clinical_outcome-01.png){fig-align="center"}
:::

## Results

#### Forest Plot

<br>

---
layout: 2-col
---

::: column
```{r eval=FALSE}
forest <- borovecki_estimates |> 
  filter(is_significant == "yes") |> 
  arrange(p.value) |> 
  slice_head(n = 30) |> 
  mutate(
    gene=fct_reorder(gene, estimate)
  ) |> 
  ggplot(aes(x=estimate, y=gene))+
  geom_errorbarh(aes(xmin = conf.low, xmax = conf.high)) +   
  geom_point() +
  labs(
    title = "Genes Associated with hungington disease",
    x = "Estimates (95%CIs)",
    y = NULL
  ) +
  theme_minimal() 

```
:::

::: column
![](images/forest-02.png){fig-align="center"}
:::

## Results

#### Volcano Plot

<br>

::: column
```{r eval=FALSE}
volcano <- ggplot(borovecki_estimates, aes(x = estimate, y = -log10(p.value))) + 
  geom_point(aes(color=is_significant, label=lbl), alpha=0.5, size=1) +
  geom_text_repel(aes(color=is_significant, label = lbl), size = 2, max.overlaps = Inf) +
  labs(
    title = "Genes Associated with hungington disease",
    subtitle = "Genes highlighted in turquoise were significant after multiple testing correction",
    x = "Estimates",
    y = "-log10(p)"
  ) +
  theme_minimal() +
  theme(legend.position = "none") 

```
:::

::: column
![](images/volcano-01.png){fig-align="center"}
:::

## Discussion

<br>

**Bioinformatics** is an **essential** tool in modern biological research.

<br>

[Data wrangling]{.underline}

:   Chaos -\> Biologically interpretable structure

[Statistical analysis + Data visualisation]{.underline}

:   Reveal key signals

<br>

**Bioinformatics** is an essential tool in modern biological research

<br> \* Data wrangling \* Chaos $\rightarrow$ Biologically interpretable structure <br> \* Statistical analysis + Data visualisation \* Reveal key signals

<br>

**Collaboration** between scientists and bioinformaticians allows for a better understanding of how **diseases** work
