---
title: "Group04 Presentation R4BDS"
author: "Eric Calaveras - Sol Escamilla - Beatriz García - Lucía López - Diego Medina"
format: revealjs
editor: visual
css: styles.css
echo: True
---

## Introduction

<div>

Dataset obtained from an article (Borovecki et al. 2005) about the correlation between the expression of different genes with the development of Huntington's disease.

Dataset content:

-   x: Probes IDs and gene expression values of each sample

-   y: Diagnosis (symptomatic, control)

The desired output to start working with the dataset is a tibble with tidy data:

1.  Each row is an observation
2.  Each column is a variable
3.  Each cell is a single value

</div>

## Materials and methods: Wrangling I: Initial tibble

<br>

With the data that we had, we created a tibble, where X is now a new column containing the samples from the microarray in order and Y is a new column containing the diagnosis of the patient in that respective sample.

Apart from that we have different columns for each sample representing all the genes from each one.

<br>

```{r eval=FALSE}
borovecki <- borovecki$x |>
  as_tibble() |>
  mutate(
    sample = row_number(), 
    .before = 1, 
    diagnosis = borovecki$y
  )
```

## Materials and methods: Wrangling II: Pivot_longer

<br>

We make use of the pivot function in order to get all the columns containing the different genes and including them all in the same column called probeID and placing the values from the different probe IDs into a new column called expressionValue.

<br>

```{r eval=FALSE}
borovecki <- borovecki |>
  pivot_longer(
    cols = ends_with("at"),
    names_to = "probeID", 
    values_to = "expressionValue",
    values_drop_na = TRUE
  )
```

## Bioconductor

<br>

The hgu133a.db package is useful to handle Affymetrix Human Genome U133A Microarrays as it allows us to do many different annotations.

In our case, thanks to the hgu133a.db package, we were able to annotate all the different probe IDs into real gene names so that the posterior work was done more easily.

<br>

```{r eval=FALSE}
#| message = FALSE
borovecki <- borovecki |>
  mutate(
    gene = mapIds(
      hgu133a.db, 
      keys=probeID, 
      keytype = "PROBEID", 
      column="SYMBOL",
      multiVals="first"
    )
  ) |>
  filter(!is.na(gene)) |>
  # I specify the package because AnnotationDbi also has a select function
  dplyr::select(-probeID)
```

## Obtaining gene-level data

Que se vea cómo varios probes de un mismo gen se colapsan en un solo valor promediando.

Explicar que en microarrays varios probes mapean a un mismo gen, reduciendo ruido y facilitando hacer estudios.

## Log2 transformation and categories

Poner en la diapo: - gráfico del antes/despues - distribución raw vs log2

Explicar: - Log2 reduce la diferencia entre valores bajos y altos. - Mejora visualización y normaliza la escala. - Explicar por qué aparecen valores negativos.

## Visualization 1

```{r, echo=FALSE}


```

## Visualization 2

## Discussion

Gracias a usar todas estas herramientas hemos podido encontrar genes mas singnificativos e importantes en esta enfermedad
