---
title: "Data processing"
format: html
---

### Load the libraries

```{r}
#| message: FALSE
library(tidyverse)
library(fs)
library(patchwork)
library(dplyr)

```

## Load the data

```{r}
borovecki_tibble <- readRDS(file = "~/projects/group04_project/data/borovecki_tibble.rds")
borovecki_estimates <- readRDS(file = "~/projects/group04_project/data/borovecki_estimates.rds")
```

## What is Huntington's disease?

Huntington's disease (HD) is an inherited disorder that causes nerve cells (neurons) in parts of the brain to gradually break down and die. The disease attacks areas of the brain that help to control voluntary (intentional) movement, as well as other areas. This disease affects areas of the brain that are associated with control of voluntary movement. The people affected by this disease develop uncontrollable movements (chorea) or instead become rigid (akinesia), in addition to that, some patients develop abnormal body postures (dystonia) and also some problems with personality disorders, emotion and behavior. The symptoms of the disease get worse over time, with them appearing typically in middle-aged people and in children, but with less frequency. When these symptoms appear, they are usually problems with balance or movement, cognitive symptoms, changes in behavior or mild clumsiness. Changes in behavior may include mood swings; feeling irritable (cranky); not being active; or feeling apathetic (uninterested), depressed, or angry. These symptoms may decrease as the disease progresses. But in some people, the symptoms can continue and may include angry outbursts, thoughts of suicide, deep depression, and psychosis (losing touch with reality). People with HD may withdrawal from social activities. When cognitive problems are severe enough that the person cannot function in daily life, the condition is described as dementia. But many people with HD stay aware of their environment and can express their emotions.

### Huntington's Disease prevalence

According to a 2022 review, the incidence of Huntington’s disease was 0.48 new cases per 100000 people in a year, and the prevalence of the disease was 4.88 cases per 100000 people. This prevalence is higher in Europe, North America and Australia than in Africa or Asia. Health-care accessibility, HTT gene haplotypes and different mutation rates may be a reason for this ethnic variation.

Also, an increase of incidence and prevalence is mentioned, due to the advancements in molecular genetic testing which induces an increase in the diagnosis of the disease. https://www.rarediseaseadvisor.com/disease-info-pages/huntington-disease-epidemiology/

### Data overview and inspections

Before applying any transformation, it is important to inspect the distribution of the raw microRNA expression values. Gene expression data often span several orders of magnitude, and this dataset is no exception. To visualize this, we generated a histogram and a boxplot of the raw expressionValue variable.

```{r}
#| message = FALSE
# Histogram of raw expression values 
raw_data1 <- ggplot(borovecki_tibble, aes(x = expressionValue)) +   
  geom_histogram(bins = 50, 
                 fill = "blue", 
                 alpha = 0.6) +   
  theme_minimal(base_size = 16) +   
  scale_x_continuous(labels = scales::comma) +   
  labs(title = "Distribution of Raw Expression Values",        
       x = "Expression value",        
       y = "Count") +
  theme(plot.title = element_text(size = 13,
                                  hjust = 0.5)) +
  plot_annotation(caption="Data from: Borovecki et al. 2005")

raw_data1

ggsave(
  filename = "~/projects/group04_project/results/raw_data1.png", 
  device = "png",                               
  dpi = 300   # Resolution
)
```

This plot shows that the raw expression values are extremely skewed. Almost all values are concentrated near zero, while one or a few values reach very high levels. This makes it nearly impossible to interpret the distribution or compare expression between genes.

```{r}
# Getting min and max values of raw data
borovecki_tibble |>
  summarise(min_exp = min(expressionValue),
            max_exp = max(expressionValue))
```

### Log transformation

Since this dataset contains microRNA expression values, the raw counts cannot be negative. A value of 0 simply indicates complete absence of expression for that particular microRNA. However, gene expression data often span several orders of magnitude, which can make the data difficult to interpret and visualize in their original scale.

To address this, we apply a log2 transformation to the expression values. This transformation serves two purposes:

1.  It normalizes the distribution, reducing the influence of extremely large values.

2.  It spreads out lower values, making subtle differences more interpretable.

After applying the log2 transformation, it is normal to obtain negative values, which correspond to expression values between 0 and 1 in the original scale. These negative values do not indicate “negative expression,” but rather reflect very low levels of expression.

```{r}
# Data transformation
borovecki_data <- borovecki_tibble |> 
  mutate(log2_expression = log2(expressionValue))
```

After transforming, we compute the minimum and maximum values:

```{r}
# Getting min and max values after log2 transformation
borovecki_data |>
  summarise(min_exp = min(log2_expression),
            max_exp = max(log2_expression))
```

```{r}
#| message = FALSE

#fill function
fill_scale <- scale_fill_manual(
  name   = NULL,
  values = c("Raw data" = "blue",
             "Log2 data" = "purple")
)
# Comparison between raw data and log2 data
raw_histogram <- ggplot(borovecki_data, aes(x = expressionValue,fill = "Raw data")) +
  geom_histogram(bins = 50, alpha = 0.6) +
  fill_scale+
  theme_minimal(base_size = 14) +
  labs(x = "Expression value",
       y = "Count") +
  theme(plot.title = element_text(size = 13, 
                                  hjust = 0.5)) 

log2_histogram <- ggplot(borovecki_data, aes(x = log2_expression,  fill = "Log2 data")) +
  geom_histogram(bins = 100, alpha = 0.6) +
  fill_scale +
  theme_minimal(base_size = 14) +
  labs(x = "log2(Expression value)",
       y = "Count") +
  theme(plot.title = element_text(size = 13,
                                  hjust = 0.5))

#Calculate max from Y axis (ExpressionValue)
#raw_hist_counts <- borovecki_data |>
#  select(expressionValue) |>
#  mutate(bin = cut(expressionValue, breaks = 50)) |>
#  count(bin, name = "count")

#max_raw <- raw_hist_counts |>
#  summarise(max(count)) |>
#  pull()

#Calculate max from Y axis (Log2_expression)
#log_hist_counts <- borovecki_data |>
#  select(log2_expression) |>
#  mutate(bin = cut(log2_expression, breaks = 100)) |>
#  count(bin, name = "count")

#max_log <- log_hist_counts |>
#  summarise(max(count)) |>
#  pull()

#Claculate commun max
#max_y <- max(max_raw, max_log)

#Add limit to the histograms
#raw_histogram  <- raw_histogram  + ylim(0, max_y)
#log2_histogram <- log2_histogram + ylim(0, max_y)


#Plot histograms
raw_histogram + log2_histogram + 
  plot_layout(guides = "collect")+
  plot_annotation(title="Comparison between Raw and Log2-Transformed Expression Data", 
                  caption="Data from: Borovecki et al. 2005") &
  theme(legend.position = "right")
  


ggsave(
  filename = "~/projects/group04_project/results/comparison1_raw_log2.png", 
  device = "png",                               
  dpi = 300   # Resolution
)
```

Zoom

```{r}
#| message = FALSE
borovecki_data |> 
  ggplot(aes(x = log2_expression)) +
  geom_histogram(bins = 100, fill = "purple", alpha = 0.6) +
  theme_minimal(base_size = 14) +
  labs(title = "Log2-transformed expression values",
       x = "log2(Expression value)",
       y = "Count") +
  theme(plot.title = element_text(size = 13,
                                  hjust = 0.5))
```

```{r}
#| message = FALSE
# Comparison between raw data and log2 data
raw_boxplot <- ggplot(borovecki_data, aes(y = expressionValue)) +
  geom_boxplot(fill = "blue", alpha = 0.6) +
  theme_minimal(base_size = 14) +
  labs(title = "Raw data",
       y = "Expression value") +
  theme(plot.title = element_text(size = 15,
                                  hjust = 0.5))

log2_boxplot <- ggplot(borovecki_data, aes(y = log2_expression)) +
  geom_boxplot(fill = "purple", alpha = 0.6) +
  theme_minimal(base_size = 14) +
  labs(title = "Transformed data",
       y = "log2(expressionValue)") +
  theme(plot.title = element_text(size = 15, 
                                  hjust = 0.5))

raw_boxplot + log2_boxplot +
  plot_annotation(title="Comparison between Raw and Log2-Transformed Expression Data", 
                  caption="Data from: Borovecki et al. 2005")

ggsave(
  filename = "~/projects/group04_project/results/comparison2_raw_log2.png", 
  device = "png",                               
  dpi = 300   # Resolution
)
```

The raw expression data is highly skewed, with most values clustered near zero and a few extremely high outliers. This makes the distribution difficult to interpret because differences among low and moderate expression levels are compressed and almost invisible. The wide dynamic range also exaggerates the influence of outliers, preventing a clear view of the underlying structure.

After log2 transformation, data becomes more balanced and easier to interpret. High values are compressed and low values are expanded, revealing clear separation between low, moderate-low, moderate-high, and high expression groups. The variance is more stable, outliers are reduced, and the transformed distribution better reflects meaningful biological differences.

### Classification according to expression level

In the next step, we created a categorical variable that groups gene expression values into six expression categories. Since the minimum and maximum observed expression levels in the dataset range from -3.3219 to 16.1313, we divided this continuous range into six intervals of equal width. Each interval represents a different level of gene expression, from the lowest category ("Very Low") to the highest one ("Very High").

```{r}
# Grouping according expression level
borovecki_grouped <- borovecki_data |> 
  mutate(expressionClass = case_when(log2_expression < -0.079717 ~ "Very Low",
    -0.079717 <= log2_expression & log2_expression < 3.162494 ~ "Low",
    3.162494 <= log2_expression & log2_expression < 6.404705 ~ "Moderate Low",
    6.404705 <= log2_expression & log2_expression < 9.646916 ~ "Moderate High",
    9.646916 <= log2_expression & log2_expression < 12.889127 ~ "High",
    12.889127 <= log2_expression ~ "Very High"))

borovecki_grouped$expressionClass <- factor(borovecki_grouped$expressionClass,
            levels = c("Very Low", "Low", "Moderate Low",                                                    "Moderate High", "High", "Very High"))
            
borovecki_grouped |> 
  select(log2_expression, expressionClass)
```

We then generated a bar plot showing how many genes fall into each category.

```{r}
#| message = FALSE
grouped_log2 <- borovecki_grouped |>
count(expressionClass) |>
  
# Visualization
ggplot(aes(x = expressionClass,
           y = n,
           fill = expressionClass)) +
  geom_col(colour = "purple",
           fill = "purple",
           alpha = 0.5) +
  theme_minimal(base_size = 16) +
  theme(axis.text.x = element_text(
        angle = 45,
        hjust = 1)) +
  labs(x = "Expression level group",
       y = "Number of genes",
       title = "Gene counts per expression category") +
  theme(plot.title = element_text(size = 15, 
                                  hjust = 0.5)) +
  plot_annotation(caption="Data from: Borovecki et al. 2005")

grouped_log2

ggsave(
  filename = "~/projects/group04_project/results/grouped_log2.png", 
  device = "png",                               
  dpi = 300   # Resolution
)
```

In typical gene expression datasets, values usually show a right-skewed distribution, where a small number of genes have extremely high expression levels. The fact that almost all data points are concentrated in the first categories, and only a tiny fraction reaches the highest levels, supports the idea that the extreme value is not representative of the general expression distribution.

Specifically, we will see how many genes are in each expression level group.

```{r}

# Get total count in each expression group and percent.
borovecki_grouped |> 
  count(expressionClass) |> 
  mutate(percent = round(100 * n / sum(n), 2))
```

We want to get the genes that are more differentially expressed and we will only filter for the significant ones (p-value \< 0.05) and also setting a filter to only select those that are more differentially expressed (estimates \> 1 or estimates \< -1)

## Data Visualization I

```{r}

Expression_value_vs_log2Expression <- ggplot(data = borovecki_tibble, aes(x= diagnosis, y = expressionValue)) +
  geom_boxplot() +

ggplot(data = borovecki_grouped, aes(x= diagnosis, y = log2_expression, color = diagnosis))+

  geom_boxplot()
Expression_value_vs_log2Expression
ggsave(
  filename = "~/projects/group04_project/results/Expression_value_vs_log2Expression.png",
  plot = Expression_value_vs_log2Expression,
  device = "png",
  dpi = 300,
)

  
```

```{r}

borovecki_tibble |>
  slice_max(expressionValue, n = 1)
```

```{r}

borovecki_tibble |>
  slice_min(expressionValue, n = 1)
```

```{r}
borovecki_grouped |>
  filter(gene == "HBA1") |>
  ggplot(aes(x = gene, y = log2_expression, color = diagnosis)) +
  geom_boxplot() +
  labs(
    title = "Expression of HBA1",
    x = "Gene",
    y = "Expression Value"
  )
```

```{r}
borovecki_grouped |>
  filter(gene == "ZNF675") |>
  ggplot(aes(x = gene, y = log2_expression, color = diagnosis)) +
  geom_boxplot() +
  labs(
    title = "Expression of ZNF675",
    x = "Gene",
    y = "Expression Value"
  )
```

```         
```

```         
```

## Data Visualization II

```{r}
library("tidyverse")
library("here")
```

```{r}
geneExpression_comparison_boxplot <- ggplot(borovecki_grouped |>
  filter(gene %in% c("HTT","FAN1", "LIG1", "MLH1", "MSH3", "PMS1", "PMS2")), aes(gene, log2_expression, fill = diagnosis)) +
  geom_boxplot() +
  theme_minimal() +
  labs(
    title = "Gene expression values in control and symptomatic individuals",
    x = "Genes",
    y = "Genes expression"
  ) +
  theme(
    plot.title = element_text(size = 12)) 

ggsave(
  filename = "~/projects/group04_project/results/geneExpression_comparison_boxplot.png",
  plot = geneExpression_comparison_boxplot,
  device = "png",
  dpi = 300,
)

```

This plot compares the expression levels of several genes that have been previously associated with Huntington's disease according to the literature. We wanted to evaluate if this is also reflected in our data set by examining differences in gene expression values between control individuals and symptomatic individuals.
The plot allows us to visually assess whether symptomatic individuals show higher or lower gene expression levels compared to control individuals.
Overall, we can observe a shift in expression values between control and symptomatic individuals, which matches what found in literature.

#### Violin plots

In order to understand better how the gene expression values change between diagnosis and control individuals, we decided to compare the data using violin plots.

```{r}
HTT_expression_violin <- ggplot(borovecki_grouped |>
filter(gene == "HTT"), aes(diagnosis, log2_expression, fill = diagnosis)) +
  geom_violin() +
  theme_minimal() +
   labs(
    title = "HTT expression distribution in control vs symptomatic individuals",
    x = "diagnosis",
    y = "genes expression"
  ) +
  theme(
    plot.title = element_text(size = 12))

ggsave(
  filename = "~/projects/group04_project/results/HTT_expression_violin.png",
  plot = HTT_expression_violin,
  device = "png",
  dpi = 300,
)


```

For HTT gene, symptomatic patientes tend to show a lower gene expression level compared to control patients. This is particularly interesting, because HTT is the key gene involved in Huntington's disease. https://www.uniprot.org/uniprotkb/P42858/entry

```{r}

PMS1_expression_violin <- ggplot(borovecki_grouped |>
filter(gene == "PMS1"), aes(diagnosis, log2_expression, fill = diagnosis)) +
  geom_violin() +
  theme_minimal() +
  labs(
    title = "PMS1 expression distribution in control vs symptomatic individuals",
    x = "diagnosis",
    y = "genes expression"
  ) +
  theme(
    plot.title = element_text(size = 12))

ggsave(
  filename = "~/projects/group04_project/results/PMS1_expression_violin.png",
  plot = PMS1_expression_violin,
  device = "png",
  dpi = 300,
)
```

For PMS1, symptomatic patients show a higher expression level compared to controls.

```{r}

ZDHHC17_expression_violin <- ggplot(borovecki_grouped |>
filter(gene == "ZDHHC17"), aes(diagnosis, log2_expression, fill = diagnosis)) +

  geom_violin() +
  theme_minimal() +
  labs(
    title = "ZDHHC17 expression distribution in control vs symptomatic individuals",
    x = "diagnosis",
    y = "genes expression"
  ) +
  theme(
    plot.title = element_text(size = 10))

ggsave(
filename = "~/projects/group04_project/results/ZDHHC17_expression_violin.png",
  plot = ZDHHC17_expression_violin,
  device = "png",
  dpi = 300,
)
```

For ZDHHC17, there is a higher gene expression for symptomatic patients compared to control patients. ZDHHC17 is a HTT-interacting protein that plays a key role in palmitoylation of neuronal proteins. https://www.uniprot.org/uniprotkb/Q8IUH5/entry

```{r}
genesExpression_geompoint <- ggplot(borovecki_grouped |>
  filter(gene %in% c("HTT","PMS1", "PMS2", "ZDHHC17", "HBA1", "ZNF675")), aes(log2_expression, gene, color = diagnosis)) +
  geom_point() + 
  theme_minimal() +
  labs(
    title = "Gene class expression distribution in control vs symptomatic individuals",
    x = "Gene expression (log 2)",
    y = "Genes"
  ) +
  theme(
    plot.title = element_text(size = 10))

ggsave(
filename = "~/projects/group04_project/results/genesExpression_geompoint.png",
  plot = genesExpression_geompoint,
  device = "png",
  dpi = 300,
)
```

### ERIC

We want to get the genes that are more differentially expressed and we will only filter for the significant ones (p-value \< 0.05) and also setting a filter to only select those that are more differentially expressed (estimates \> 1 or estimates \< -1)

```{r}
borovecki_estimates_filtered <- borovecki_estimates |> 
filter(is_significant == "yes")
```

We pull the genes (filtered ones and full dataset) and create two .txt documents so that we can copy and paste in ShinyGO and see the enrichment.

```{r}
genes <- borovecki_estimates_filtered |> pull(gene)

writeLines(genes, "genesfilter.txt")

genesall <- borovecki_estimates |> pull (gene)

writeLines(genesall, "genesall.txt")
```

### Data Visualization I

```{r}
borovecki_data <- borovecki_data |> 
  mutate(log2_expression = log2(expressionValue))
```

```{r}
borovecki_data |>
  filter(gene == "HBA1") |>
  ggplot(aes(x = gene, y = log2_expression)) +
  geom_boxplot() +
  labs(
    title = "Expression of HBA1",
    x = "Gene",
    y = "Expression Value"
  )
```

```{r}
borovecki_data |>
  filter(gene == "ZNF675") |>
  ggplot(aes(x = gene, y = log2_expression)) +
  geom_boxplot() +
  labs(
    title = "Expression of ZNF675",
    x = "Gene",
    y = "Expression Value"
  )
```

Repair genes expression level boxplot

### HTT

<https://doi.org/10.1093/braincomms/fcae418>

Huntington's disease is caused by CAG repeat in the HTT gene. Recent genome-wide Association studies (GWASs) have identified genetic modifiers that influence average age of onset in the disease, among the modifiers several genes related to DNA mismatch repair *FAN1*, *LIG1*, *MLH1*, *MSH3*, *PMS1* and *PMS2*,7 which are thought to impact the somatic expansion of the CAG repeat in vulnerable neural cell types.

```{r}
#| message = FALSE
dnarepair_genes <- c("PMS1", "FAN1", "LIG1", "MLH1", "MSH3", "PMS2")

dnarepair_data<- borovecki_data|>
  filter(gene %in% dnarepair_genes)


Expression_of_DNA_repair_genes_vs_clinical_outcome <- ggplot(dnarepair_data, aes(x = log2_expression, 
             y = gene, 
             fill = diagnosis))+
  geom_boxplot(alpha = 0.4, outlier.shape = NA)+
  labs(
    x = "Expression level",
    y = "Genes",
    title = "Expression of DNA repair genes vs clinical outcome",
    caption="Data from: Borovecki et al. 2005"
  ) +
  theme_minimal() +
  theme(legend.position = "right ")


ggsave(
  filename = "~/projects/group04_project/results/Expression_of_DNA_repair_genes_vs_clinical_outcome.png",
  plot = Expression_of_DNA_repair_genes_vs_clinical_outcome,
  device = "png",
  dpi = 300,
)
             
```

```{r}
genes_compare<-borovecki_data|>
  filter(gene %in% c("ZNF675","HBA1"))

genes_wider <- genes_compare|>
  select(sample, gene, log2_expression) |>
  pivot_wider(names_from = gene, values_from = log2_expression)

ggplot(genes_wider, aes(x = ZNF675, y = HBA1)) +
  geom_point(alpha = 0.6) +
  geom_smooth(method = "lm") +
  labs(
    title = "Comparison of HBA1 vs ZNF675 Expression",
    x = "ZNF675 expression",
    y = "HBA1 expression"
  ) +
  theme_minimal()


```

```{r}
genes_relevant0 <- borovecki_data |>
  filter(gene %in% c("HTT", "LIG1"))

genes_wide4 <- genes_relevant0 |>
  select(sample, gene, log2_expression) |>
  pivot_wider(names_from = gene, values_from = log2_expression)

ggplot(genes_wide4, aes(x = HTT, y = LIG1)) +
  geom_point(alpha = 0.6) +
  geom_smooth() +
  labs(
    title = "Comparison of HTT vs LIG1 Expression",
    x = "HTT expression",
    y = "LIG1 expression"
  ) 

```

```{r}
#| message = FALSE
genes_relevant1 <- borovecki_data |>
  filter(gene %in% c("HTT", "MLH1"))

genes_wide1 <- genes_relevant1 |>
  select(sample, gene, log2_expression) |>
  pivot_wider(names_from = gene, values_from = log2_expression)

Comparison_of_HTT_vs_MHL1_Expression <- ggplot(genes_wide1, aes(x = HTT, y = MLH1)) +
  geom_point(alpha = 0.6) +
  geom_smooth() +
  labs(
    title = "Comparison of HTT vs MHL1 Expression",
    x = "HTT expression",
    y = "MHL1 expression",
    caption="Data from: Borovecki et al. 2005"
  ) 

Comparison_of_HTT_vs_MHL1_Expression 

ggsave(
  filename = "~/projects/group04_project/results/Comparison_of_HTT_vs_MHL1_Expression.png",
  plot = Comparison_of_HTT_vs_MHL1_Expression,
  device = "png",
  dpi = 300,
)

```

```{r}
#| message = FALSE
genes_Relevant <- borovecki_data |>
  filter(gene %in% c("HTT", "PMS1"))

genes_wide3 <- genes_Relevant |>
  select(sample, gene, log2_expression) |>
  pivot_wider(names_from = gene, values_from = log2_expression)

Comparison_of_HTT_vs_PMS1_Expression <- ggplot(genes_wide3, aes(x = HTT, y = PMS1)) +
  geom_point(alpha = 0.6) +
  geom_smooth() +
  labs(
    title = "Comparison of HTT vs PMS1 Expression",
    x = "HTT expression",
    y = "PMS1 expression",
    caption="Data from: Borovecki et al. 2005"
  ) 
Comparison_of_HTT_vs_PMS1_Expression 
ggsave(
  filename = "~/projects/group04_project/results/Comparison_of_HTT_vs_PMS1_Expression.png",
  plot = Comparison_of_HTT_vs_PMS1_Expression,
  device = "png",
  dpi = 300,
)
```

There is a significant change in the Data Visualization I part I and Data Visualization I part II. This is because we have applied log2 normalization in gene expression so the data is compressed allowing a better comparison between genes and a more normal-like distribution. In the second plots we lose outliers influence, the variation becomes more uniform and the trend line becomes more stable. All this improvements reflect the biological trends more reliably.

MHL1, PMS1, LIG1 are all DNA mismatch repair gene, helping the replication errors and maintain genome integrity. We want to analyse what happens to these genes expression compared to HTT.

Biologically the relationship between HTT vs PMS1. PMS1 is a DNA is not perfectly linear. At a low HTT expression, PMS1 expression is high at first but then as HTT increases PMS1 expression drops sharply. When the HTT is in a medium expression level the PMS1 tends to increase, it could mean that as the HTT is more expressed the PMS1 reacts to handle the damage. Finally PMS1 expression decreases as HTT is at its highest, this could mean that the repair gene has been overwhelm by the HTT.

The biological relationship between HTT vs MHL1 we can see a similarity with the PSM1 plot. At first the expression of MHL1 is high but it decrease when HTT's expression is low as the HTT increases the MHL1 increase but then when the HTT expression level its high the MHL1 shows a gradual decreases. This results, as the PMS1 before, suggest an impaired mismatch repair capacity when the HTT is elevated.

Finally the relationship between HTT vs LIG1, this plot is shows a slightly positive trend, as the HTT expression level increases, LIG1 increases slowly. But the slope is weak ans the grey region, the interval, is wide meaning the relationship is weak, this could be due to the HTT expression and how it might influence the this DNA repair gene.

```{r}
genes_repair <- borovecki_data |>
  filter(gene %in% c("HTT", "PMS1", "FAN1", "LIG1", "MLH1", "MSH3", "PMS2"))

genes_wide5 <- genes_repair |>
  select(sample, gene, log2_expression) |>
  pivot_wider(names_from = gene, values_from = log2_expression)

genes_long<- genes_wide5|>
  pivot_longer(
    cols = c(PMS1, FAN1, LIG1, MLH1, MSH3, PMS2),
    names_to = "reapir_genes",
    values_to = "log2_exp"
  )|>
  drop_na(HTT, log2_exp)



ggplot(genes_long, aes(x = HTT, y = log2_exp)) +
  geom_point(alpha = 0.6) +
  geom_smooth() +
  labs(
    title = "Comparison of HTT vs Repair gene Expression",
    x = "HTT expression",
    y = " Repair gene Expression"
  ) 
```

The plot above has all the relevant DNA mismatch repair genes mentioned on the article "Genomic characterization of Huntington’s disease genetic modifiers informs drug target tractability" combined into one single table where the curve summarizes the overall trend across all repair genes and compare them to HTT expression. the plot shows a very slight negative trend. The overall conclusion is that the HTT expression does not strongly influence increasing or decreasing the DNA mismatch repair genes, weak relationship.

After doing the process on ShinyGO, we see no important fold enrichment on this case.
